crypto/cert_manager.go

```diff
@@ -106,8 +106,8 @@ func (c *certManager) Verify(hostname string) error {
 
 	var opts x509.VerifyOptions
 	if c.config != nil {
+		// we don't need to care about the tls.Config.ServerName here, since hostname has already been set to that value in the session setup
```

Comment:
I still don't really understand why you are doing this change… If the config is not nil, you don't set DNSName at all, not even to hostname, right? How is the verification supposed to work then?